### ISSUE TITLE
[OpenAI] Add support for Batch API

### ISSUE THREAD

**Issue by mssfang** [Azure/azure-sdk-for-java#39773]
https://platform.openai.com/docs/api-reference/batch/create

TBD: We need to wait for AOAI TypeSpec definition to be ready.

**Comment by imsosleepy** [comment]
Can I handle this issue?

Based on my research, it seems that this issue needs to be added to https://github.com/Azure/azure-rest-api-specs. Specifically, a model for batch/create should be added in the path /specification/cognitiveservices/OpenAI.Inference.

Additionally, can I freely add this issue to https://github.com/Azure/azure-rest-api-specs?

**Comment by mssfang** [comment]
@imsosleepy Thank you for reaching out to us.
Yes. We are welcome to contribute to the repo. You are free to open PRs and issue to the repo.

**Comment by imsosleepy** [comment]
@mssfang 
I've been analyzing the openAI part of the azure-rest-api-specs for the past few days, and it's a bit complicated and difficult for an Azure open source newbie like me to get started.... 

According to my analysis, the Files API of openAI should be implemented first to complete the batch API spec, but it seems that the [Files API part](https://platform.openai.com/docs/api-reference/files/create) is not written yet. 

So I think I need to work on the Files API, but where do I start? Should I create an issue for the Files API in this repository first?

And I was hoping you could give me some advice on how to write it in the api spec. It seems like I have to do a buzzing to add a new API SPEC, but is there a set date for the buzzing? I ask because it seems like the criteria for creating openAI versions is the 1st and 15th of each month.

**Comment by mssfang** [comment]
Hi @imsosleepy I can share the OpenAI TSP owners (@jpalvarezl @trrwilson) for more information about how to contribute to OpenAI TSP.

**Comment by imsosleepy** [comment]
@mssfang 

I think the changes to the files API are included in this PR: https://github.com/Azure/azure-rest-api-specs/pull/29067

The above issue doesn't seem to be necessary.

Using the information about the files API in the specification/ai path, I think I can write a spec for the Batch API. 

The sequence for external contributors to contribute, as I understand it, is as follows

1. Write the batch API as a TypeSpec in the correct path in the https://github.com/Azure/azure-rest-api-specs repository (this would be cognitiveservices/OpenAI.Inference)
2. Create an issue in the https://github.com/Azure/azure-rest-api-specs repository.
3. Auto-generate a Dataplane (I haven't gotten this far yet, so I'm not sure) (https://azure.github.io/typespec-azure/docs/next/howtos/DataPlane%20Generation%20-%20DPG/00howtogen)
4. Create a PR in the azure-rest-api-specs repository with the work done.
5. When that PR is merged and a draft is created, auto-generate a PR in azure-java-sdk
6. Once the PR is merged into azure-java-sdk, deploy on both sides.

I think TSP means Traning Service Provider, so I'll ask a question, here's what I'm curious about.

1. is the order I have written for outsiders to contribute correct? 
2. the code in the OpenAI.Inference/examples path looks like generated code, how was it created?
3. within the data-plane, there is code generated in the authoring path, why is it separated?

I think I'm having a bit of a problem with what I've understood and what the team is working on, but you're right that I'm in need of some help 😢.

**Comment by imsosleepy** [comment]
@mssfang Is there any way I can contribute?

**Comment by mssfang** [comment]
Hi, @imsosleepy. Thank you for your contribution. It is valuable to us for sure. While I usually don't work on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs) side, I recommend opening a new PR directly in the repo in the future. I've also heard that Azure OpenAI service will start supporting `Batch` soon in this `2024-07-01` version. Some people will be adding Batch REST APIs along with other enhancements or new features in that release. 

After that is merged, I will start to use it to generate the Java OpenAI SDK baseline and look into to support the Batch from both of Azure OpenAI and OpenAI services.

**Comment by imsosleepy** [comment]
@mssfang It seems that TypeSpec definitions have an internal timeline, and it's not yet clear where I can start contributing, but maybe monitoring the progress of this issue will help me contribute to this repository in the future?

**Comment by mssfang** [comment]
@imsosleepy Thank you for the understanding. Yes. I will keep you posted here when there is a new update.

**Comment by mssfang** [comment]
@imsosleepy Batch API is supported in the new version 
https://repo1.maven.org/maven2/com/azure/azure-ai-openai/1.0.0-beta.11/
